pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -0,0 +1,327 @@
+	}
+
+	if gardenlet.Bootstrap != nil {
+		validValues := []string{string(seedmanagement.BootstrapServiceAccount), string(seedmanagement.BootstrapToken), string(seedmanagement.BootstrapNone)}
+		if !utils.ValueExists(string(*gardenlet.Bootstrap), validValues) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("bootstrap"), *gardenlet.Bootstrap, validValues))
+		}
+	}
+
+	return allErrs
+}
+
+func validateGardenletUpdate(newGardenlet, oldGardenlet *seedmanagement.Gardenlet, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if newGardenlet.Config != nil && oldGardenlet.Config != nil {
+		configPath := fldPath.Child("config")
+
+		// Convert new gardenlet config to an internal version
+		newGardenletConfig, err := helper.ConvertGardenletConfig(newGardenlet.Config)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(configPath, newGardenlet.Config, fmt.Sprintf("could not convert gardenlet config: %v", err)))
+			return allErrs
+		}
+
+		// Convert old gardenlet config to an internal version
+		oldGardenletConfig, err := helper.ConvertGardenletConfig(oldGardenlet.Config)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(configPath, oldGardenlet.Config, fmt.Sprintf("could not convert gardenlet config: %v", err)))
+			return allErrs
+		}
+
+		// Validate gardenlet config update
+		allErrs = append(allErrs, validateGardenletConfigurationUpdate(newGardenletConfig, oldGardenletConfig, configPath)...)
+	}
+
+	// Ensure bootstrap is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.Bootstrap, oldGardenlet.Bootstrap, fldPath.Child("bootstrap"))...)
+
+	// Ensure merge with parent is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.MergeWithParent, oldGardenlet.MergeWithParent, fldPath.Child("mergeWithParent"))...)
+
+	return allErrs
+}
+
+func validateGardenletDeployment(deployment *seedmanagement.GardenletDeployment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if deployment.ReplicaCount != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*deployment.ReplicaCount), fldPath.Child("replicaCount"))...)
+	}
+	if deployment.RevisionHistoryLimit != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*deployment.RevisionHistoryLimit), fldPath.Child("revisionHistoryLimit"))...)
+	}
+
+	if deployment.Image != nil {
+		allErrs = append(allErrs, validateImage(deployment.Image, fldPath.Child("image"))...)
+	}
+
+	allErrs = append(allErrs, metav1validation.ValidateLabels(deployment.PodLabels, fldPath.Child("podLabels"))...)
+	allErrs = append(allErrs, apivalidation.ValidateAnnotations(deployment.PodAnnotations, fldPath.Child("podAnnotations"))...)
+
+	return allErrs
+}
+
+func validateImage(image *seedmanagement.Image, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if image.PullPolicy != nil {
```

Comment:
Can you also validate that `Repository` and `Tag` are not empty if set?

Reply:
OK

Reply:
Done